lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -91,7 +92,7 @@ def initialize(info = {})
     register_autofilter_services(%W{ http https })
 
     # Initialize an empty cookie jar to keep cookies
-    self.cookie_jar = Set.new
+    self.cookie_jar = HTTP::CookieJar.new
```

Comment:
As discussed on a chat, let's investigate how a wrapper would work:
```suggestion
    self.cookie_jar = Exploit::Remote::HttpCookieJar.new
```

Reply:
This will make it easier to unit tests too 🎉